Write sections on invalid commits and access control (#261)
mlswg/mls-architecture: draft-ietf-mls-architecture.md

As an example, there could be an "ordering server" Delivery Service that
broadcasts all messages received to all users and ensures that all clients see
messages in the same order. This would allow clients to only apply the first
valid Commit for an epoch and ignore subsequent ones. Clients that send a Commit
would then wait to apply it until it's broadcast back to them by the Delivery

> Service, assuming they don't receive another Commit first.

Reviewer: it is broadcast...
they do not receive...

> would then wait to apply it until it's broadcast back to them by the Delivery
> Service, assuming they don't receive another Commit first.

Reviewer 2: ```suggestion
would then wait to apply it until it is broadcast back to them by the Delivery
Service, assuming they do not receive another Commit first.
```

prevent this from causing forks.

## Invalid Commits

Situations can arise where a malicious or buggy client sends a Commit that is
not accepted by all members of the group, and the DS is not able to detect this
and reject the Commit.  For example, a buggy client might send an encrypted
Commit with an invalid set of proposals, or a malicious client might send a
malformed Commit of the form described in {{Section 16.12 of RFC9420}}.

In situations where the DS is attempting to filter redundant Commits, the DS
might update its internal state under the assumption that a Commit has succeeded
and thus end up in a state inconsistent with the members of the group.  For
example, the DS might think that the current epoch is now `n+1` and reject any
commits from other epochs, while the members think the epoch is `n`, and as a
result, the group is stuck -- no member can send a Commit that the DS will
accept.

Such “desynchronization” problems can arise even when the Delivery Service takes
no stance on which Commit is "correct" for an epoch. The DS can enable clients
to choose between Commits, for example by providing Commits in the order

> received when there are multiple, and allow clients to reject any Commits that

Reviewer 2: ```suggestion
received when and allow clients to reject any Commits that
```

I don't think "when there are multiple" does anything here.

> received when there are multiple, and allow clients to reject any Commits that

Author: ```suggestion
received and allow clients to reject any Commits that
```

violate their view of the group's policies. As such, all honest and
correctly-implemented clients will arrive at the same "first valid Commit" and
choose to process it. Malicious or buggy clients that process a different Commit
will end up in a forked view of the group.

When these desynchronizations happen, the application may choose to take action
to restore the functionality of the group.  These actions themselves can have
security implications.  For example, a client developer might have a client
automatically rejoin a group, using an external join, when it processes an
invalid Commit.  In this operation, however, the client trusts that the
GroupInfo provided by the DS faithfully represents the state of the group, and

> not, say, an earlier state containing a compromised leaf node. Even worse, the

Reviewer 2: ```suggestion
not, say, an earlier state containing a compromised leaf node. In addition, the
```
